src/NuGet.Core/NuGet.Protocol/Providers/RepositorySignatureResourceProvider.cs

```diff
@@ -95,5 +98,13 @@ public RepositorySignatureResourceProvider()
 
             return null;
         }
+
+        private static string GenerateCacheKey(ServiceIndexEntry serviceEntry)
+        {
+            var index = serviceEntry.Type.IndexOf('/');
+            var version = serviceEntry.Type.Substring(index + 1).Trim();
```

Comment:
This won't work on service index like this:

```
{
"@type": "RepositorySignatures/Versioned",
"clientVersion": "4.7.0",
 ...
},
{
"@type": "RepositorySignatures/Versioned",
"clientVersion": "5.0.0",
 ...
},
```

Reply:
Out of curiosity, why doesn't the HTTP cache not use the URL (or some string derived from the URL) as the cache key?

Reply:
Using /Versioned is a good idea, but I think it should be a separate issue/PR.

Not sure why URL's weren't used.  That decision was made >3 years ago.